File: data/multilang_comparison/program_julia/program_julia/src/func42.jl
include("program_julia_head.jl")
function func42(vars::JArray_param, PATH0::UInt64, loopsFactor::Int)::JArray
   pCounter = length(vars.data)
   if (PATH0 & 1) != 0
      params0 = JArray_param{Int}()
      params0.size = 0
      array29 = func43(params0, loopsFactor);
      array29.refC -= 1
      if array29.refC == 0
          empty!(array29.data)
          array29 = nothing
      end
   
   else 
      params0 = JArray_param{Int}()
      params0.size = 0
      array31 = func44(params0, get_path(), loopsFactor);
      array31.refC -= 1
      if array31.refC == 0
          empty!(array31.data)
          array31 = nothing
      end
   end
   params0 = JArray_param{Int}()
   params0.size = 0
   array35 = func45(params0, loopsFactor);
   return array35;
end

File: data/multilang_comparison/program_go/program_go/src/func42.go
package main
import "math/rand"
var _ = rand.Intn
func Func42(vars *ArrayParam, PATH0 uint64 , loopsFactor int) *Array{
   var pCounter int = vars.Size;
   _ = pCounter
   if (PATH0 & 1) != 0 {
      var params0 ArrayParam
      params0.Size = 0
      params0.Data = make([]*Array, params0.Size)
      var array29 *Array = Func43(&params0, loopsFactor);
      params0.Data = nil
      params0.Size = 0
      array29.RefC--
      if array29.RefC == 0 {
          array29.Data = nil
          array29 = nil
      }
   
   } else {
      var params0 ArrayParam
      params0.Size = 0
      params0.Data = make([]*Array, params0.Size)
      var array31 *Array = Func44(&params0, getPath(), loopsFactor);
      params0.Data = nil
      params0.Size = 0
      array31.RefC--
      if array31.RefC == 0 {
          array31.Data = nil
          array31 = nil
      }
   }
   var params0 ArrayParam
   params0.Size = 0
   params0.Data = make([]*Array, params0.Size)
   var array35 *Array = Func45(&params0, loopsFactor);
   params0.Data = nil
   params0.Size = 0
   return array35;
}

